### PR TITLE
Auto-Increment_version_csvlogger

### LINKED ISSUE #17886: Auto increment version for CSVLogger likeTensorBoardLogger in Fabric
### Description & Motivation

TensorBoardLogger auto increments next version, but CSVLogger doesn't, which I believe it's inconsistent.
```python
from lightning.fabric.loggers import TensorBoardLogger

logger = TensorBoardLogger("throw_away")
logger.log_metrics({"loss": 0.235, "acc": 0.75})
logger.finalize("success")
``` 
When running twice, it will create `version_0` then `version_1` folders.

However, 
```python
from lightning.fabric.loggers import CSVLogger

logger = CSVLogger("throw_away")
logger.log_metrics({"loss": 0.235, "acc": 0.75})
logger.finalize("success")
```

will give off the warning: /opt/conda/lib/python3.10/site-packages/lightning/fabric/loggers/csv_logs.py:186: UserWarning: Experiment logs directory throw_away/lightning_logs/version_0 exists and is not empty. Previous log files in this directory will be deleted when the new ones are saved!
  rank_zero_warn(

### Pitch

Given the following code, 
```python
from lightning.fabric.loggers import CSVLogger

logger = CSVLogger("throw_away")
logger.log_metrics({"loss": 0.235, "acc": 0.75})
logger.finalize("success")
```
When running this code twice (or again), the logger should create the next `version_{}` folder instead of warning.

### Alternatives

_No response_

### Additional context

_No response_

cc @borda @carmocca @justusschock @awaelchli

### PR DESCRIPTION
**Pull request by ManoBharathi93**
## What does this PR do?
       
 Multiple versions of CSVLogger will be saved in the corresponding directory (lightning logs) as similar as  Tensorboard logger. Previously it will keep the only new version and discard the old version. 

Structure of the folder:
image

Fixes #17886

           

<details>
  <summary><b>Before submitting</b></summary>

-  Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [X] Did you make sure to **update the documentation** with your changes? (if necessary) - modified docstrings
- [x] Did you write any **new necessary tests**? (not for typos and docs) 
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [] Is this pull request ready for review? (if not, please submit in draft mode)
- [] Check that all items from **Before submitting** are resolved
- [] Make sure the title is self-explanatory and the description concisely explains the PR
- [] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

**Comment by ManoBharathi93** [comment]
@awaelchli  How can I pass checks ?

File: src/lightning/fabric/loggers/csv_logs.py [hunk 1]
```diff
@@ -59,6 +63,7 @@ def __init__(
         name: str = "lightning_logs",
         version: Optional[Union[int, str]] = None,
         prefix: str = "",
+        sub_dir: Optional[_PATH] = None,
```
Comment (awaelchli) [inline src/lightning/fabric/loggers/csv_logs.py:66]: Could you open a separate issue for this and only keep the fix for the auto-increment in this PR? Then we can merge the fix faster. Thanks

Comment (ManoBharathi93) [reply src/lightning/fabric/loggers/csv_logs.py:66]: This PR is actually to fix these bug #17886.  This works well in locally. It fails  to pass in PR test cases

Comment (awaelchli) [reply src/lightning/fabric/loggers/csv_logs.py:66]: To fix the auto-increment problem, it shouldn't be necessary to introduce new API like `sub_dir`. I'm simply asking if we could have separate PRs for these two things.

Comment (awaelchli) [reply src/lightning/fabric/loggers/csv_logs.py:66]: It would be easier for me to help with this PR if it was focused on the fix.

Comment (ManoBharathi93) [reply src/lightning/fabric/loggers/csv_logs.py:66]: Sorry for delayed reply.

Actually Problem this issue arise with hidden test checks in GitHub. Better I will close this pull request so you can work on it..

**Pull request closed without merging**